src/utils.c

```diff
@@ -439,6 +450,8 @@ int descend(QPTPool_t *ctx, const size_t id, void *args,
             child.root_parent = work->root_parent;
             child.pinode = inode;
 
+            child_ed.parent_fd = d_fd;
```

Comment:
Maybe set `parent_fd` to -1 here, and set it to `d_fd` later, when non-directories are being processed. Right now, the `parent_fd` looks like it can be invalid when it is passed to subdirectories and it takes a bit more reading to see that the parent never uses it.

Reply:
Makes sense, done